src/main/java/net/coderbot/iris/mixin/fabulous/MixinOption.java

```diff
@@ -15,6 +16,6 @@
 	@Redirect(method = "method_18554", at = @At(value = "INVOKE", target = "Lcom/mojang/blaze3d/platform/GlStateManager;supportsGl30()Z"))
 	private static boolean iris$onAttemptedToSelectFabulousGraphics() {
 		// Returning false here will cause Minecraft to cycle between Fancy and Fast, disabling Fabulous graphics
-		return false;
+		return !Iris.getIrisConfig().areShadersEnabled();
```

Comment:
Instead of unconditionally returning true when shaders are enabled, we should call `supportsGl30()` first.

Reply:
Should be good now.